src/chart/generateCategoricalChart.tsx

```diff
@@ -752,11 +752,15 @@ const calculateOffset = (
     offset = appendOffsetOfLegend(offset, graphicalItems, props, prevLegendBBox);
   }
 
+  const offsetWidth = width - offset.left - offset.right;
+  const offsetHeight = height - offset.top - offset.bottom;
+
   return {
     brushBottom,
     ...offset,
-    width: width - offset.left - offset.right,
-    height: height - offset.top - offset.bottom,
+    // never return negative values for height and width
+    width: offsetWidth > 0 ? offsetWidth : 0,
```

Comment:
This works fine, sometimes you can see this written as `Math.max(offsetWidth, 0)`

Reply:
Done - that reads a little easier